src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -525,9 +531,9 @@ public override bool VisitNamedType(INamedTypeSymbol symbol)
                         _builder.Append('.');
                     }
 
-                    _builder.Append(EncodeName(symbol.Name));
+                    _builder.Append(EncodeName(symbol.IsExtension ? symbol.ExtensionGroupingName : symbol.Name));
 
-                    if (symbol.TypeParameters.Length > 0)
+                    if (symbol.TypeParameters.Length > 0 && !symbol.IsExtension)
                     {
                         _builder.Append('`');
```

Comment:
>builder.Append('`'); [](http://example.com/codeflow?start=25&length=20)

Are we supposed to add this only when ```MangleName``` property is `true`? Perhaps instead of having this `if` we should use `MetadataName` instead of `Name` above?